webapp/channels/src/components/admin_console/filter/team_filter_dropdown/team_filter_checkbox.tsx

```diff
@@ -11,36 +8,37 @@ type Props = {
     updateOption: (checked: boolean, name: string) => void;
 };
 
-class TeamFilterCheckbox extends React.PureComponent<Props> {
-    toggleOption = () => {
-        const {checked, id, updateOption} = this.props;
-        updateOption(!checked, id);
-    };
+const TeamFilterCheckbox: React.FC<Props> = ({
+    id,
+    name,
+    checked,
+    label,
+    updateOption,
+}) => {
+    const [isChecked, setIsChecked] = useState(checked);
```

Comment:
You are changing the logic here. The `useState` is not needed here, the `checked` value is handled outside the component and passed as a propperty, and updated with the `updateOption` property. If the previous existing component didn't have any `State`, is probably because the `useState` in the functional component is not needed.